rs/cli/src/unit_tests/replace.rs

```diff
@@ -0,0 +1,345 @@
+
+                // Insert mandatory features
+                for feature in &[NodeFeature::NodeProvider, NodeFeature::DataCenter, NodeFeature::DataCenterOwner] {
+                    map.insert(feature.clone(), "Some value".to_string());
+                }
+
+                map
+            },
+        },
+    }
+}
+
+fn subnet(id: u64, nodes: &[Node]) -> DecentralizedSubnet {
+    DecentralizedSubnet {
+        id: PrincipalId::new_subnet_test_id(id),
+        nodes: nodes.to_vec(),
+        added_nodes_desc: vec![],
+        removed_nodes_desc: vec![],
+        comment: None,
+        run_log: vec![],
+    }
+}
+
+fn cordoned_feature(feature: NodeFeature, value: &str) -> NodeFeaturePair {
+    NodeFeaturePair {
+        feature,
+        value: value.to_string(),
+    }
+}
+
+fn test_pretty_format_response(response: &Result<SubnetChangeResponse, anyhow::Error>) -> String {
+    match response {
+        Ok(r) => format!(
+            r#"Response was OK!
+    Added nodes:
+{},
+    Removed nodes:
+{},
+    Feature diff:
+{}
+            "#,
+            r.added_with_desc
+                .iter()
+                .map(|(id, desc)| format!("\t\t- principal: {}\n\t\t  desc: {}", id, desc))
+                .join("\n"),
+            r.removed_with_desc
+                .iter()
+                .map(|(id, desc)| format!("\t\t- principal: {}\n\t\t  desc: {}", id, desc))
+                .join("\n"),
+            r.feature_diff
+                .iter()
+                .map(|(feature, diff)| format!(
+                    "\t\t- feature: {}\n{}",
+                    feature,
+                    diff.iter()
+                        .map(|(value, (in_nodes, out_nodes))| format!("\t\t\t- value: {}, In: {}, Out: {}", value, in_nodes, out_nodes))
+                        .join("\n")
+                ))
+                .join("\n")
+        ),
+        Err(r) => format!("Response was ERR: {}", r),
+    }
+}
+
+fn pretty_print_node(node: &Node, num_ident: usize) -> String {
+    format!(
+        "{}- principal: {}\n{}  dfinity_owned: {}\n{}  features: [{}]",
+        "\t".repeat(num_ident),
+        node.id,
+        "\t".repeat(num_ident),
+        node.dfinity_owned,
+        "\t".repeat(num_ident),
+        node.features
+            .feature_map
+            .iter()
+            .map(|(feature, value)| format!("({}, {})", feature, value))
+            .join(", ")
+    )
+}
+
+fn pretty_print_world(available_nodes: &[Node], subnet: &DecentralizedSubnet) -> String {
+    format!(
+        r#"Available nodes:
+{}
+Observed subnet:
+    - id: {}
+      nodes:
+{}"#,
+        available_nodes.iter().map(|node| pretty_print_node(node, 1)).join("\n"),
+        subnet.id,
+        subnet.nodes.iter().map(|node| pretty_print_node(node, 2)).join("\n")
+    )
+}
+
+#[test]
+fn should_skip_cordoned_nodes() {
+    // World setup
+    let available_nodes = vec![
+        node(1, true, &[(NodeFeature::Country, "Country 1"), (NodeFeature::City, "City 1")]),
```

Comment:
We're getting rid of the "NodeFeature::City" since it clashes with the adopted target topology. Can you use DC instead?